hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java

```diff
@@ -51,13 +51,27 @@ public static boolean isEmpty(@Nullable final Key pbjKey) {
             return true;
         }
         if (pbjKey.hasKeyList()) {
-            return !((KeyList) key.value()).hasKeys()
-                    || (((KeyList) key.value()).hasKeys()
-                            && ((KeyList) key.value()).keys().isEmpty());
+            final var keyList = (KeyList) key.value();
+            if (!keyList.hasKeys() || keyList.keys().size() == 0) {
+                return true;
+            }
+            for (final var k : keyList.keys()) {
+                if (!isEmpty(k)) {
+                    return false;
+                }
+            }
+            return true;
         } else if (pbjKey.hasThresholdKey()) {
-            return !((ThresholdKey) key.value()).hasKeys()
-                    || (((ThresholdKey) key.value()).hasKeys()
-                            && ((ThresholdKey) key.value()).keys().keys().isEmpty());
+            final var thresholdKey = (ThresholdKey) key.value();
+            if ((!thresholdKey.hasKeys() || thresholdKey.keys().keys().size() == 0)) {
```

Comment:
is it possible for `thresholdKey.keys().keys()` to be null?

Reply:
yeah it can be.. changed to `thresholdKey.keys().keysOrElse(Collections.emptyList()).size() == 0)`